test/BaselineOutput/Common/OVA/OVA-TrainTest-iris-out.txt

```diff
@@ -1,33 +1,30 @@
-maml.exe TrainTest test=%Data% tr=OVA{p=AvgPer{ lr=0.8 }} norm=No dout=%Output% data=%Data% out=%Output% seed=1 xf=Term{col=Label}
+maml.exe TrainTest test=%Data% tr=OVA{p=AvgPer{ lr=0.8 } useprob=-} norm=No dout=%Output% data=%Data% out=%Output% seed=1 xf=Term{col=Label}
 Not adding a normalizer.
 Training learner 0
-Training calibrator.
 Training learner 1
-Training calibrator.
 Training learner 2
-Training calibrator.
 Not training a calibrator because it is not needed.
 
 Confusion table
           ||========================
 PREDICTED ||     0 |     1 |     2 | Recall
 TRUTH     ||========================
         0 ||    50 |     0 |     0 | 1.0000
-        1 ||     0 |    46 |     4 | 0.9200
+        1 ||    30 |     0 |    20 | 0.0000
         2 ||     0 |     0 |    50 | 1.0000
           ||========================
-Precision ||1.0000 |1.0000 |0.9259 |
-Accuracy(micro-avg): 0.973333
-Accuracy(macro-avg): 0.973333
-Log-loss:           0.291849
-Log-loss reduction: 0.734347
+Precision ||0.6250 |0.0000 |0.7143 |
+Accuracy(micro-avg): 0.666667
+Accuracy(macro-avg): 0.666667
+Log-loss:           17.052120
+Log-loss reduction: -14.521509
```

Comment:
Can you benchmark this PR before committing? All of the unit tests indicate a large drop in metrics.

Reply:
See the comment I made to Wei-Shing above. The drop is also there in master when not using a calibrated learner. Part of this PR makes it so that you have to pass in a calibrated learner if you want one and the OneVersusAll model no longer auto creates one for you. Due to this, I had to change the tests to not use a calibrated learner as well and therefor the drop in metrics. This can be reproduced in master by following the steps listed above.

Reply:
This demonstrates you can match the same low score. Can the new version match the quite higher scores of the old version? 

I would recommend modifying the unit tests so that the old high scores are maintained.

Reply:
The problem is that now its a 2 step process. You have to make create the calibrated trainer, and then use that in the OneVersusAll trainer. In this particular case, the test is using the command line parameters `p=AvgPer...` and I couldn't see a way to use the command line parameters to use a calibrated trainer instead of a non-calibrated trainer. Do you know if we have it set up to where we can do it? If so, could you point me in the right direction for that? I looked, and from what I saw it didn't look like we had the option to specify a calibrated trainer directly from the command line.

Reply:
That sounds like a breaking change for MAML users (mostly internal users). I do not think it's possible to calibrate a trainer within OVA in MAML, besides telling OVA to calibrate it for you. Is there a way to keep the MAML interface as it was (and achieve the goals of this PR)?

Speaking of, what is the goal of the PR? Seems we're just making it very hard for users to get good scores, whereas the current behavior simply gives them a good score by default.

Reply:
The goal of this PR is to have the OVAModelParameters strongly typed so that the users can access the underlying models. Right now they are cast as an array of object, so the user cannot do anything with them without knowing their type and casting it. This PR strongly types OVAModelParameters and anything that uses that so the model can be accessed directly. I will add unit tests with a calibrated trainer showing the results are the same as what is there currently.

Reply:
Sounds like a good goal. It'll be great to see unit tests showing the same good scores as before.  Thanks for all your work.

Is there a way to keep the same interface/behavior for MAML folks?

This change may also affect NimbusML. They may have their own OVA, at which point they would be unaffected.